### PR TITLE
fix(manager): get rid of whitelisting for sctool restore cmd parameters

### PR DESCRIPTION
**Pull request by mikliapko**
Complements https://github.com/scylladb/scylla-manager/issues/3950

Changes:
- Simplified restore extra parameters handling approach - all params now should be passed in SCT_MGMT_RESTORE_EXTRA_PARAMS var. Motivation - the list of possible restore parameters are quite long and handling them all as separate instance introduces some complexity. In addition to that, some of them were introduced only for testing purposes or even exist only in test builds the jobs for which we trigger in manual mode only.
- Thanks to param extra_environment_variables, mgmt_restore_params removed from managerPipeline.

### Testing

- [x] https://jenkins.scylladb.com/job/scylla-staging/job/mikita/job/manager-master/job/backup-restore-custom-dataset-test-PR/12/

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### CODE REVIEW

**Comment by mikliapko** [comment]
@scylladb/qa-maintainers Could someone please take a look?

File: docs/configuration_options.md [hunk 1]
```diff
@@ -362,7 +362,6 @@
 | **<a href="#user-content-use_capacity_reservation" name="use_capacity_reservation">use_capacity_reservation</a>**  | reserves instances capacity for whole duration of the test run (AWS only).<br>Fallbacks to next availabilit zone if capacity is not available | N/A | SCT_USE_CAPACITY_RESERVATION
 | **<a href="#user-content-bisect_start_date" name="bisect_start_date">bisect_start_date</a>**  | Scylla build date from which bisecting should start.<br>Setting this date enables bisection. Format: YYYY-MM-DD | N/A | SCT_BISECT_START_DATE
 | **<a href="#user-content-bisect_end_date" name="bisect_end_date">bisect_end_date</a>**  | Scylla build date until which bisecting should run. Format: YYYY-MM-DD | N/A | SCT_BISECT_END_DATE
-| **<a href="#user-content-mgmt_restore_params" name="mgmt_restore_params">mgmt_restore_params</a>**  | Manager restore operation specific parameters: batch_size, parallel | N/A | SCT_MGMT_RESTORE_PARAMS
```
Comment (soyacz) [inline docs/configuration_options.md:365]: mgmt_restore_extra_params is missing here. Worth to recreate this file with `hydra update-conf-docs`

**Review by soyacz — DISMISSED** [review]
small issue, LGTM

File: docs/configuration_options.md (hunk 1 above)
Comment (mikliapko) [reply docs/configuration_options.md:365]: Fixed

File: vars/managerPipeline.groovy [hunk 2]
```diff
@@ -177,8 +172,8 @@ def call(Map pipelineParams) {
                  description: (
                      'Extra environment variables to be set in the test environment, uses the java Properties File Format.\n' +
                      'Example:\n' +
-                     '\tSCT_STRESS_IMAGE.cassandra-stress=scylladb/cassandra-stress:3.13.0\n' +
-                     '\tSCT_N_DB_NODES=6\n' +
+                     '\tSCT_MGMT_RESTORE_EXTRA_PARAMS=--batch-size 50 --parallel 0\n' +
```
Comment (fruch) [inline vars/managerPipeline.groovy:175]: There no need to update this one on every change, few examples are enough.

I would be too long to hold all the possibilities

**Review by fruch — APPROVED** [review]

**Review by soyacz — APPROVED** [review]

**Pull request merged**